sql/core/src/test/scala/org/apache/spark/sql/execution/RowFormatConvertersSuite.scala

```diff
@@ -58,6 +58,41 @@ class RowFormatConvertersSuite extends SparkPlanTest with SharedSQLContext {
     assert(!preparedPlan.outputsUnsafeRows)
   }
 
+  test("coalesce can process unsafe rows") {
+    val plan = Coalesce(1, outputsUnsafe)
+    val preparedPlan = sqlContext.prepareForExecution.execute(plan)
+    assert(getConverters(preparedPlan).size === 1)
+    assert(preparedPlan.outputsUnsafeRows)
+  }
```

Comment:
Sorry, I am not sure if I understand your question. `Coalesce` only have a single child. I am not sure what mixed rows mean. Could you give me one example? Thank you!